crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,287 @@
+        min_len: 24,
+    },
+    SensitivePattern {
+        pattern: Pattern::Regex(&AWS_API_KEY_REGEX),
+        comment: "AWS API Key",
+        min_len: 16,
+    },
+    SensitivePattern {
+        pattern: Pattern::Regex(&HEROKU_API_KEY_REGEX),
+        comment: "Heroku API Key",
+        min_len: 12,
+    },
+    SensitivePattern {
+        pattern: Pattern::Regex(&PASSWORD_IN_URL_REGEX),
+        comment: "Password in URL",
+        min_len: 14,
+    },
+    SensitivePattern {
+        pattern: Pattern::Regex(&GOOGLE_SERVICE_ACCOUNT_REGEX),
+        comment: "Google (GCP) Service-account",
+        min_len: 14,
+    },
+    SensitivePattern {
+        pattern: Pattern::Regex(&TWILIO_API_KEY_REGEX),
+        comment: "Twilio API Key",
+        min_len: 32,
+    },
+    SensitivePattern {
+        pattern: Pattern::Contains("-----BEGIN RSA PRIVATE KEY-----"),
+        comment: "RSA Private Key",
+        min_len: 64,
+    },
+    SensitivePattern {
+        pattern: Pattern::Contains("-----BEGIN OPENSSH PRIVATE KEY-----"),
+        comment: "SSH (OPENSSH) Private Key",
+        min_len: 64,
+    },
+    SensitivePattern {
+        pattern: Pattern::Contains("-----BEGIN DSA PRIVATE KEY-----"),
+        comment: "SSH (DSA) Private Key",
+        min_len: 64,
+    },
+    SensitivePattern {
+        pattern: Pattern::Contains("-----BEGIN EC PRIVATE KEY-----"),
+        comment: "SSH (EC) Private Key",
+        min_len: 64,
+    },
+    SensitivePattern {
+        pattern: Pattern::Contains("-----BEGIN PGP PRIVATE KEY BLOCK-----"),
+        comment: "PGP Private Key Block",
+        min_len: 64,
+    },
+];
+
+static mut MIN_PATTERN_LEN: Option<usize> = None;
+static INIT: Once = Once::new();
+
+fn get_min_pattern_len() -> usize {
+    unsafe {
+        INIT.call_once(|| {
+            MIN_PATTERN_LEN = Some(
+                SENSITIVE_PATTERNS
+                    .iter()
+                    .map(|pattern| pattern.min_len)
+                    .min()
+                    .unwrap_or(0),
+            );
+        });
+        MIN_PATTERN_LEN.unwrap_or(0)
+    }
+}
+
+fn is_high_entropy(text: &str) -> bool {
+    let entropy = calculate_shannon_entropy(text);
+    entropy > 4.5 // TODO: Make this optional, or controllable
+}
+
+/// Inspired by https://github.com/nickdeis/eslint-plugin-no-secrets/blob/master/utils.js#L93
+/// Adapted from https://docs.rs/entropy/latest/src/entropy/lib.rs.html#14-33
+/// Calculates Shannon entropy to measure the randomness of data. High entropy values indicate potentially
+/// secret or sensitive information, as such data is typically more random and less predictable than regular text.
+/// Useful for detecting API keys, passwords, and other secrets within code or configuration files.
+fn calculate_shannon_entropy(data: &str) -> f64 {
+    let mut freq = [0usize; 256];
+    let mut len = 0usize;
+    for &byte in data.as_bytes() {
+        freq[byte as usize] += 1;
+        len += 1;
+    }
```

Comment:
We could directly get the slice length here:
```suggestion
    let mut len = data.len();
    for &byte in data.as_bytes() {
        freq[byte as usize] += 1;
    }
```